app/api/route/schemas/user_schemas.py

```diff
@@ -1,34 +1,41 @@
-from typing import Any
+from typing import Any, Mapping
 
 import marshmallow
 from apiflask import fields
 from marshmallow import fields as marshmallow_fields
 
-from api.db.models.user_models import Role, RoleType
+from api.db.models import user_models
+from api.route.schemas import request_schema
+from api.services import users
```

Comment:
I'm curious if you wrote this manually, or your IDE formats it like this? I've always disliked this approach for importing multiple objects (unless it's a very large number) as it works against writing code quickly

Reply:
oh i was just following [google's python import guidelines](https://google.github.io/styleguide/pyguide.html#22-imports), i think it often improves readability to see the class/function in the context of the module (e.g. `module.class`) especially if the class/function has a name that's a bit generic i.e. not entirely obvious what context it's for, and so i think something like `user_service.CreateUserParams` is clear that the CreateUserParams is part of the user_service module. otherwise it can potentially be confusing since we also have `user_models.User` in the `db.models` package.

i used to do something like `from api.services.user_service import CreateUserParams` but i've (very recently) started to appreciate the formality of google's guideline.

as an aside, [in this comment on jesse's PR](https://github.com/navapbc/template-application-flask/pull/56#discussion_r1060731741) i noticed that by importing the module rather than specific items in the module, it also helps avoid accidentally re-exporting things that weren't meant to be re-exported, which can avoid importing the wrong thing from the wrong module

Reply:
I guess I don't "get" those guidelines. It looks more like a consistency reason rather than for a purpose especially when it adds a lot of verbosity with little gain. Maybe I'm just stuck in my ways, but it feels like more effort (since the IDE doesn't inherently do it) and more confusing to me (since every reference to a class/function requires a much longer signature). Not saying there aren't scenarios where I like the pattern, but it doesn't feel right

Reply:
I see that being a potential issue, although I've only seen it when refactoring (a function/class used to be in a file, got moved elsewhere, and imports daisy-chain without being noticed).

I wonder if there's a flake8 or other linter that can catch that? I found https://pypi.org/project/flake8-import-graph/ - but that seems to require explicit rules (eg. model files can't import sqlalchemy)